pandas/core/window.py

```diff
@@ -247,13 +235,12 @@ def _prep_values(self, values=None, kill_inf=True):
                     "cannot handle this type -> {0}" "".format(values.dtype)
                 )
 
-        if kill_inf:
-            values = values.copy()
-            values[np.isinf(values)] = np.NaN
+        # Always convert inf to nan
+        values[np.isinf(values)] = np.NaN
```

Comment:
You missed a copy call here. Results in https://github.com/kernc/backtesting.py/issues/14 on write-protected arrays.